### PR TITLE
FEATURE: when suggesting usernames skip input that consist entirely of disallowed characters

### PR DESCRIPTION
**Pull request by AndrewPrigorshnev**
We used to do this before:

```ruby
suggested_username = UserNameSuggester.suggest(
  external_username || external_name || external_email
)
```
The problem with this is that if `external_username` is made entirely of disallowed characters, this code just falls back to the `fallback_username = "user"` instead of trying to use `external_name` and `external_email`.

This PR changes this logic in a way, so we'll be always trying other options in such cases.

I've implemented it in `UsernameSuggester` itself, to avoid writing this logic again and again in every place where we need it. But it was pretty hard to choose a good name for the new method. Sadly, I couldn't just change the signature of existing method:

```ruby
# the old signature:
def self.suggest(name_or_email,  current_username = nil)
...
end

# the new signature could look like this (with a '*'), 
# but this won't work because the second parameter has default value:
def self.suggest(*name_or_email,  current_username = nil)
...
end
```

Eventually, I decided to create a new method:
```ruby
# The input parameter is an array here.
# I tried to come up with something that would emphatize 
# that we use array as an unput here, but something like `suggest_from_arr`
# doesn't sound good. I think we can migrate our code to this new method 
# and then remove the old `suggest()` method. We'll end up with the `suggest_username` 
# method then, which is a bit wordy, but not so bad after all
def self.suggest_username(input,  current_username = nil)
```

I still feel like this is not an ideal solution. If somebody has ideas about a better way of solving this, I'd love to hear them.

### CODE REVIEW

**Comment by AndrewPrigorshnev** [comment]
@gschlager a keyword argument works, thank you! I've pushed a fix. Happy that we don't have to create the second method here.

**Review by davidtaylorhq — APPROVED** [review]
Looks good to me. We'll need to double check for any custom plugins that may be overriding `Auth::Result#username_suggester_attributes`, but I don't think there will be many
